semantic/unet/utils.py

```diff
@@ -73,3 +74,129 @@ def overlay_labels_on_input(img: Image, labels : np.ndarray, alpha : float = 0.4
     labels_img.putalpha(floor(255 * alpha))
     print(img.size, labels_img.size)
     return Image.alpha_composite(img, labels_img)
+
+
+def get_image_with_high_res_center(model, img : Image, zoom : float = 2, output_dir: str = "output") -> np.ndarray:
+    # Create the output directory if it does not exist
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
+
+    # Get input size of network
+    input_size = model.layers[0].get_output_at(0).get_shape().as_list()
```

Comment:
make use of `infer` so you don't have to repeat this step throughout with tracking `input_size`